File: send-message/php/send_message.php
<?php
/**
 * SendSeven API - Send Message Example (PHP)
 *
 * Demonstrates how to send a text message using the SendSeven API.
 */

// Load environment variables from .env file (simple implementation)
function loadEnv(string $path): void {
    if (!file_exists($path)) {
        return;
    }
    $lines = file($path, FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES);
    foreach ($lines as $line) {
        if (strpos(trim($line), '#') === 0) {
            continue;
        }
        if (strpos($line, '=') !== false) {
            list($name, $value) = explode('=', $line, 2);
            putenv(trim($name) . '=' . trim($value));
        }
    }
}

loadEnv(__DIR__ . '/.env');

// Configuration from environment
$API_TOKEN = getenv('SENDSEVEN_API_TOKEN');
$TENANT_ID = getenv('SENDSEVEN_TENANT_ID');
$API_URL = getenv('SENDSEVEN_API_URL') ?: 'https://api.sendseven.com/api/v1';
$CONVERSATION_ID = getenv('CONVERSATION_ID');

/**
 * Send a text message to a conversation.
 *
 * @param string $conversationId The UUID of the conversation
 * @param string $text The message text to send
 * @return array The created message object
 * @throws Exception If the API request fails
 */
function sendMessage(string $conversationId, string $text): array {
    global $API_TOKEN, $TENANT_ID, $API_URL;

    $url = $API_URL . '/messages';

    $payload = json_encode([
        'conversation_id' => $conversationId,
        'text' => $text,
        'message_type' => 'text',
    ]);

    $ch = curl_init($url);
    curl_setopt_array($ch, [
        CURLOPT_POST => true,
        CURLOPT_POSTFIELDS => $payload,
        CURLOPT_RETURNTRANSFER => true,
        CURLOPT_HTTPHEADER => [
            'Authorization: Bearer ' . $API_TOKEN,
            'X-Tenant-ID: ' . $TENANT_ID,
            'Content-Type: application/json',
        ],
    ]);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    $error = curl_error($ch);
    curl_close($ch);

    if ($error) {
        throw new Exception("cURL Error: $error");
    }

    if ($httpCode >= 400) {
        throw new Exception("API Error $httpCode: $response");
    }

    return json_decode($response, true);
}

// Main execution
function main(): void {
    global $API_TOKEN, $TENANT_ID, $CONVERSATION_ID;

    // Validate configuration
    if (!$API_TOKEN) {
        echo "Error: SENDSEVEN_API_TOKEN environment variable is required\n";
        exit(1);
    }

    if (!$TENANT_ID) {
        echo "Error: SENDSEVEN_TENANT_ID environment variable is required\n";
        exit(1);
    }

    if (!$CONVERSATION_ID) {
        echo "Error: CONVERSATION_ID environment variable is required\n";
        exit(1);
    }

    echo "Sending message to conversation: $CONVERSATION_ID\n";

    try {
        $message = sendMessage(
            $CONVERSATION_ID,
            'Hello from the SendSeven PHP SDK! 🐘'
        );

        echo "Message sent successfully!\n";
        echo "  ID: {$message['id']}\n";
        echo "  Status: {$message['status']}\n";
        echo "  Created at: {$message['created_at']}\n";
    } catch (Exception $e) {
        echo "Error: " . $e->getMessage() . "\n";
        exit(1);
    }
}

main();

File: send-message/python/send_message.py
#!/usr/bin/env python3
"""
SendSeven API - Send Message Example

Demonstrates how to send a text message using the SendSeven API.
"""

import os
import requests
from dotenv import load_dotenv

# Load environment variables from .env file
load_dotenv()

# Configuration from environment
API_TOKEN = os.getenv("SENDSEVEN_API_TOKEN")
TENANT_ID = os.getenv("SENDSEVEN_TENANT_ID")
API_URL = os.getenv("SENDSEVEN_API_URL", "https://api.sendseven.com/api/v1")
CONVERSATION_ID = os.getenv("CONVERSATION_ID")


def send_message(conversation_id: str, text: str) -> dict:
    """
    Send a text message to a conversation.

    Args:
        conversation_id: The UUID of the conversation
        text: The message text to send

    Returns:
        dict: The created message object

    Raises:
        requests.HTTPError: If the API request fails
    """
    url = f"{API_URL}/messages"

    headers = {
        "Authorization": f"Bearer {API_TOKEN}",
        "X-Tenant-ID": TENANT_ID,
        "Content-Type": "application/json",
    }

    payload = {
        "conversation_id": conversation_id,
        "text": text,
        "message_type": "text",
    }

    response = requests.post(url, json=payload, headers=headers)
    response.raise_for_status()

    return response.json()


def main():
    # Validate configuration
    if not API_TOKEN:
        print("Error: SENDSEVEN_API_TOKEN environment variable is required")
        return

    if not TENANT_ID:
        print("Error: SENDSEVEN_TENANT_ID environment variable is required")
        return

    if not CONVERSATION_ID:
        print("Error: CONVERSATION_ID environment variable is required")
        return

    print(f"Sending message to conversation: {CONVERSATION_ID}")

    try:
        message = send_message(
            conversation_id=CONVERSATION_ID,
            text="Hello from the SendSeven Python SDK! 🐍"
        )

        print("Message sent successfully!")
        print(f"  ID: {message['id']}")
        print(f"  Status: {message['status']}")
        print(f"  Created at: {message['created_at']}")

    except requests.HTTPError as e:
        print(f"API Error: {e.response.status_code}")
        print(f"Response: {e.response.text}")
    except Exception as e:
        print(f"Error: {e}")


if __name__ == "__main__":
    main()
